src/tools/msvc.jam

```diff
@@ -907,7 +925,7 @@ local rule configure-really ( version ? : options * )
             # version from the path.
             # FIXME: We currently detect both Microsoft Visual Studio 9.0 and
             # 9.0express as 9.0 here.
-            if [ MATCH "(Microsoft Visual Studio 15)" : $(command) ]
+            if [ MATCH "(14.10)" : $(command) ]
```

Comment:
I know but with 2017 that string is not a part of the path. Only predictable string is `14.10`